### ISSUE TITLE
Use onLoad instead of onResolve

### ISSUE THREAD

**Issue by unlocomqx** [koluch/esbuild-plugin-sass#18]
It's better to use `onLoad` because when `onResolve` is used, it doesn't allow other plugins to handle resolving the files.
In my case, I have a scss file that is imported like so
```ts
import "@scss/main.scss"
```
I resolve it using the [alias plugin](https://github.com/igoradamenko/esbuild-plugin-alias#readme), but the resolve callback is only called once and not chained through all the plugins

So the code would look like this

```js
build.onLoad({ filter: /.\.(scss|sass)$/, namespace: "file" }, async (args) => {
    let css = (await sassRender({ file: args.path })).css.toString();
    return {
        contents: css,
        loader: 'css',
        watchFiles: [args.path],
    };
});
```

**Comment by unlocomqx** [comment]
A potential advantage of this approach is that css urls will be handled by esbuild loaders, so there's no need to replace them

**Comment by koluch** [comment]
Hi @unlocomqx,

Sorry, but I think you are wrong. I am using `onResolve` callback exactly because I want to make it possible to chain plugins. The result of this plugin is a compiled css file, which will be handled with default behaviour, including other files, setup to handle css files.

Your approach suggests to return inline CSS content, which will not be handled by other plugins.

Maybe I am wrong and missing something in documentation?

**Comment by unlocomqx** [comment]
Hi, I checked the esbuild sources, it returns after the first "successful" onResolve, so it doesn't really allow chaining.

Call to resolve function
https://github.com/evanw/esbuild/blob/785d8c372c1e2ae03cac43a6fddc1129a03562ed/lib/shared/common.ts#L766

Break when there is a result
https://github.com/evanw/esbuild/blob/785d8c372c1e2ae03cac43a6fddc1129a03562ed/lib/shared/common.ts#L800

In my case, I was trying to use the alias plugin with your plugin but that didn't work so I had to alter the code.

The returned compiled css is passed to the esbuild css loader to handle it, it doesn't necessarily mean it's inline css
The css loader is able to bundle it to a file and handle the relative urls and so on
https://esbuild.github.io/content-types/#css

I also just checked the less plugin code and it uses the onLoad event and the css loader
https://github.com/iam-medvedev/esbuild-plugin-less/blob/master/src/index.ts

I'm not sure if that's confirmation to my idea though because the plugins I checked use a mix of both methods.

Finally, it's not a critical issue, and could be a matter of opinion. The docs didn't explicitly prohibited it anyway.
